### PR TITLE
functions/slack: message verification - tokens -> signatures

### PR DESCRIPTION
**Pull request by ace-n**
~~**Do not merge** until cl/284093789 (related docs changes) has approval~~

### CODE REVIEW

**Comment by leahecole** [comment]
@ace-n I know that the appengine/datastore CI errors have been fixed - could you pretty please resolve the conflicts on this when you get a chance so we can try to merge this in?

**Comment by ace-n** [comment]
@leahecole I'll work on getting the tests passing. In the meantime, could you (or someone else) review + LGTM?

**Comment by ace-n** [comment]
#fyi failures are unrelated to this PR.

File: functions/slack/main.py [hunk 1]
```diff
@@ -26,13 +28,24 @@
...
 # [START functions_verify_webhook]
-def verify_web_hook(form):
-    if not form or form.get('token') != config['SLACK_TOKEN']:
+# Python 3+ version of https://github.com/slackapi/python-slack-events-api/blob/master/slackeventsapi/server.py
+def verify_signature(request):
+    timestamp = request.headers.get('X-Slack-Request-Timestamp', '')
+    signature = request.headers.get('X-Slack-Signature', '')
+
+    req = str.encode('v0:' + str(timestamp) + ':') + request.get_data()
+    request_hash = 'v0=' + hmac.new(
```
Comment (ace-n) [inline functions/slack/main.py]: @crwilcox any ideas on other ways to do this? (Maybe a format string, or moving the concat to line 48?)

Comment (crwilcox) [reply functions/slack/main.py]: I think `request.headers` should already be a `dict(str:str)` so the cast doesn't seem necessary. I think format strings might also be preferable to concatenation as well?

The encode could also be moved to the `timestamp =` call but probably doesn't matter much?

```
    req = 'v0:{}:{}'.format(str.encode(timestamp), request.get_data())
    hash = hmac.new(
        str.encode(config['SLACK_SECRET']),
        req, hashlib.sha256
    ).hexdigest()
    request_hash = 'v0={}'.format(hash)
```

**Pull request merged**